app/src/main/java/com/example/trainboard/MainActivity.kt

```
@@ -1,47 +1,566 @@
+        WhiteContainer(modifier = Modifier.height(130.dp)){
+                Column(
+                    modifier = Modifier.padding(15.dp).fillMaxWidth(),
+
+
                     )
+                {
+                        Text("${viewModel.trainInfo?.outboundJourneys!![0].originStation.displayName} -> ${viewModel.trainInfo?.outboundJourneys!![0].destinationStation.displayName}",
+                            fontWeight = FontWeight.Bold,
+                            fontSize = 20.sp)
+
+
+                        Text("${getDateFromUTC(viewModel.trainInfo?.outboundJourneys!![0].departureTime)}                       Adults: ${viewModel.trainInfo?.numberOfAdults}, Children: ${viewModel.trainInfo?.numberOfChildren}",
+
+                            fontSize = 15.sp)
+
+                }
+        }
+
+        for (journey in viewModel.trainInfo?.outboundJourneys!!)
+        {
+            WhiteContainer (
+                modifier = Modifier.height(100.dp)
+
+            ){
+                Column (
+                    modifier = Modifier.padding(15.dp).fillMaxWidth(),
+                    verticalArrangement = Arrangement.Center,
+                    horizontalAlignment = Alignment.CenterHorizontally
+                        )
+                {
+
+                    Row{
+                        Text("${getTimeFromUTC(journey.departureTime)} → ${getTimeFromUTC( journey.arrivalTime)} " ,
+                            fontWeight = FontWeight.Bold,
+                            fontSize = 20.sp)
+                        Text("           ${journey.journeyDurationInMinutes.div(60) }h ${journey.journeyDurationInMinutes%60}mins")
+                    }
+
+                    var totalPrice = 0
+                    try{
+                        totalPrice = journey.tickets.minOf{it.priceInPennies}/100
+                    }
+                    catch(_: Exception){
+
```

Fair enough for ignoring an exception for a quick project like ours, but maybe at least print something? Alternatively, use `.runCatching` or a similar function?